File: js/cm-api-v2/PostTaxonomyGroup.js
// DocSection: cm_api_v2_post_taxonomy_group
// Tip: Find more about JS/TS SDKs at https://docs.kontent.ai/javascript
// Using ES6 syntax
import { ManagementClient } from '@kentico/kontent-management';

const client = new ManagementClient({
  projectId: '<YOUR_PROJECT_ID>',
  apiKey: '<YOUR_API_KEY>'
});

client.addTaxonomy()
  .withData(
    {
      name: "Personas",
      externalId: "Tax-Group-123",
      codename: "people",
      terms: [
        {
          name: "Coffee expert",
          externalId: "Tax-term-456",
          codename: "expert",
          terms: [
            {
              name: "Barista",
              externalId: "Tax-term-789",
              terms: []
            },
            {
              name: "Cafe owner",
              externalId: "Tax-term-101",
              terms: []
            }
          ]
        },
        {
          name: "Coffee enthusiast",
          codename: "enthusiast",
          externalId: "Tax-term-112",
          terms: [
            {
              name: "Coffee lover",
              codename: "lover",
              externalId: "Tax-term-131",
              terms: []
            },
            {
              name: "Coffee blogger",
              codename: "blogger",
              externalId: "Tax-term-145",
              terms: []
            }
          ]
        }
      ]
    }
  )
  .toObservable()
  .subscribe((response) => {
    console.log(response);
  },
  (error) => {
    console.log(error);
  });
// EndDocSection

File: net/cm-api-v2/PostTaxonomyGroup.cs
// DocSection: cm_api_v2_post_taxonomy_group
// Tip: Find more about .NET SDKs at https://docs.kontent.ai/net
using Kentico.Kontent.Management;

var client = new ManagementClient(new ManagementOptions
{
    ApiKey = "<YOUR_API_KEY>",
    ProjectId = "<YOUR_PROJECT_ID>"
});

var response = await client.CreateTaxonomyGroupAsync(new TaxonomyGroupCreateModel
{
    Name = "Personas",
    ExternalId = "Tax-Group-123",
    Codename = "people",
    Terms = new TaxonomyTermCreateModel[]
    {
        new TaxonomyTermCreateModel
        {
            Name = "Coffee expert",
            Codename = "expert",
            ExternalId = "Tax-term-456",
            Terms = new TaxonomyTermCreateModel[]
            {
                new TaxonomyTermCreateModel
                {
                    Name = "Barista",
                    ExternalId = "Tax-term-789",
                    Terms = Enumerable.Empty<TaxonomyTermCreateModel>()
                },
                new TaxonomyTermCreateModel
                {
                    Name = "Cafe owner",
                    ExternalId = "Tax-term-101",
                    Terms = Enumerable.Empty<TaxonomyTermCreateModel>()
                }
            }
        },
        new TaxonomyTermCreateModel
        {
            Name = "Coffee enthusiast",
            Codename = "enthusiast",
            ExternalId = "Tax-term-112",
            Terms = new TaxonomyTermCreateModel[]
            {
                new TaxonomyTermCreateModel
                {
                    Name = "Coffee lover",
                    ExternalId = "Tax-term-131",
                    Codename = "lover",
                    Terms = Enumerable.Empty<TaxonomyTermCreateModel>()
                },
                new TaxonomyTermCreateModel
                {
                    Name = "Coffee blogger",
                    ExternalId = "Tax-term-145",
                    Codename = "blogger",
                    Terms = Enumerable.Empty<TaxonomyTermCreateModel>()
                }
            }
        }
    }
});
// EndDocSection